MaxText/checkpointing.py

```diff
@@ -58,11 +60,40 @@ def create_orbax_checkpoint_manager(
   return mngr
 
 
+def _find_idx(array: np.ndarray, replica_axis_idx: int):
+  """Returns the index along given dimension that the current host belongs to."""
+  idx = None
+  for idx, val in np.ndenumerate(array):
+    if val.process_index == jax.process_index():
+      break
+  return idx[replica_axis_idx]
+
+
+def _replica_devices(device_array: np.ndarray, replica_axis_idx: int):
+  """Returns the devices from the replica that current host belongs to.
+
+  Replicas are assumed to be restricted to the first axis.
+
+  Args:
+    device_array: devices of the mesh that can be obtained by mesh.devices()
+    replica_axis_idx: axis dimension along which replica is taken
+
+  Returns:
+    devices inside the replica that current host is in
+  """
+  idx = _find_idx(device_array, replica_axis_idx)
+  replica_result = np.take(device_array,
+                           idx,
+                           axis=replica_axis_idx)
+  return np.expand_dims(replica_result, axis=replica_axis_idx)
```

Comment:
Should there be a safety here to detect if this host is part of different replicas? Seems good to detect it here.

Reply:
Since the check is happening in orbax, I will skip it herehttps://github.com/google/orbax/blob/4d7702f8966a97e6ad0e2a44afe19e95e8df76b4/checkpoint/orbax/checkpoint/type_handlers.py#L1591

we can revisit if we find it useful.